### PR TITLE
netcoredbg: 2.0.0-895 -> 2.2.0-961

### PR DESCRIPTION
**Pull request by konradmalik**
###### Description of changes

See [my update request](https://github.com/NixOS/nixpkgs/issues/217720).

This description is edited, removed my brainstorming and added a list of things done/changed:
- does not work out-of-the-box on aarch64-* due to https://github.com/dotnet/runtime/issues/78286 but the included patch (arm64.patch) fixes those compilation issues correctly
- coreclr is now in dotnet/runtime repo (using 7.x branch as per netcoredbg recommendation)
- netcoredbg now depends (during runtime) on DbgShim which was moved to a separate repo (dotnet/diagnostic). We need to pass `-p:UseDbgShimDependency=true` in order to include it in a list of deps.
- needed to enable selfContainedBuild for the managed part since it seems like it's the easiest way to force dotnet to copy relevant binaries to the out folder as well (like libdbgshim.so retrieved via the abovementioned nuget dependency)
- still, the program did not work at this stage (on linux only). Running `ldd` showed that libdbgshim cannot find `libstdc++.so.6`
- `runtimeDeps` of buildDotnetModule cannot help here, as it wraps dotnet executables, but our executable is build via cMake so it's not available for that patching.
- I've settled on using `autoPatchelfHook` to add `stdenv.cc.cc.lib` to binaries, which seems to work

Btw. Since I've already figured out a bunch of things about this package and I'm bound to dotnet via my current job and to netcoredbg via using NeoVim for dev - is there a way to become a co-maintainer on this package? I'd happily help to keep it up-to-date and working.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by mdarocha — APPROVED** [review]
LGTM :)

If you want to add yourself as a maintainer, you just change and commit (best as a separate commit in this PR) `maintainers/maintainer-list.nix` and then the `maintainer` field of this package.

As far as I know, we don't really have any official "becoming a maintainer process". Adding yourself to that field will cause ofborg to notify you of future PRs to this packages, and your email will be visible as maintainer in search.nixos.org.

Also if your PR won't get merged for a while, you can try posting it to https://discourse.nixos.org/t/prs-ready-for-review/3032 (and try to review other PRs posted there in exchange ;) )

**Comment by konradmalik** [comment]
thanks for the info regarding the discourse thread, didn't know this existed 😄 
Also, just added myself to maintainers.

File: pkgs/development/tools/misc/netcoredbg/default.nix [hunk 1]
```diff
@@ -49,19 +47,32 @@ let
     inherit pname version src dotnet-sdk;
 
     projectFile = "src/managed/ManagedPart.csproj";
+    # nix run .#legacyPackages.x86_64-linux.netcoredbg.passthru.fetch-deps
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/netcoredbg/default.nix:50]: Ideally this should be put into the update script. dotnet has a generic one which should be extended.

**Comment by SuperSandro2000** [comment]
Can you put all the commits required for the update together and put the extra information into the commit message? Right now it looks like that if any of those commits would be missing, it wouldn't work.

File: pkgs/development/tools/misc/netcoredbg/default.nix (hunk 1 above)
Comment (konradmalik) [reply pkgs/development/tools/misc/netcoredbg/default.nix]: added a simple updateScript, tested and seems to work ok

**Review by mdarocha — APPROVED** [review]

File: pkgs/development/tools/misc/netcoredbg/default.nix [hunk 2]
```diff
@@ -1,45 +1,44 @@
...
 
-  dotnet-sdk = dotnetCorePackages.sdk_6_0;
+  dotnet-sdk = dotnetCorePackages.sdk_7_0;
 
   src = fetchFromGitHub {
     owner = "Samsung";
     repo = pname;
     rev = version;
-    sha256 = "sha256-zOfChuNjD6py6KD1AmN5DgCGxD2YNH9gTyageoiN8PU=";
+    sha256 = hash;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/netcoredbg/default.nix:21]:
```suggestion
    inherit hash;
```

File: pkgs/development/tools/misc/netcoredbg/default.nix [hunk 3]
```diff
@@ -51,28 +50,42 @@ let
...
-    ln -s $out/share/netcoredbg/netcoredbg $out/bin/netcoredbg
+    cp ./lib/netcoredbg/* $out/share/netcoredbg
+    # darwin won't work unless we link all files
+    ln -s $out/share/netcoredbg/* "$out/bin/"
   '';
 
   passthru = {
     inherit (managed) fetch-deps;
+
+    updateScript = [ ./update.sh pname version meta.homepage ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/netcoredbg/default.nix:81]: I am not sure if that works or not

**Pull request merged**